pkg/asset/ignition/bootstrap/bootstrap_test.go

```diff
@@ -0,0 +1,90 @@
+package bootstrap
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBootstrapLoad(t *testing.T) {
+	tests := []struct {
+		name       string
+		config     []byte
+		wantFound  bool
+		fetchErr   error
+		wantErrMsg string
+	}{
+		{
+			name:      "valid",
+			wantFound: true,
+			config:    []byte(`{"ignition":{"config":{},"security":{"tls":{}},"timeouts":{},"version":"2.2.0"},"networkd":{},"passwd":{},"storage":{},"systemd":{}}`),
+		},
+		{
+			name:       "no version",
+			wantErrMsg: "error: invalid config version",
+			config:     []byte(`{"ignition":{"config":{},"security":{"tls":{}},"timeouts":{},"version":""},"networkd":{},"passwd":{},"storage":{},"systemd":{}}`),
+		},
+		{
+			name:       "empty",
+			wantErrMsg: "failed to unmarshal bootstrap.ign",
+			config:     []byte(""),
+		},
+		{
+			name:       "custom fetch error",
+			fetchErr:   fmt.Errorf("this test"),
+			wantErrMsg: "this test",
+			config:     []byte(""),
+		},
+		{
+			name:       "custom fetch error",
+			fetchErr:   fmt.Errorf("this test"),
+			wantErrMsg: "this test",
+			config:     []byte(""),
+		},
+		{
+			name:     "file not found error",
+			fetchErr: os.NewSyscallError("open", syscall.ENOENT),
+			config:   []byte(""),
+		},
+		{
+			name: "custom content error",
+			config: []byte(`{"ignition":{"config":{},"security":{"tls":{}},"timeouts":{},"version":"2.2.0"},"networkd":{},"passwd":{},
+			"storage":{"files":[
+				{"filesystem":"root","path":"/etc/myfile","user":{"name":"root"},"append":true,
+				"contents":{"source":"data:text/plain;charset=utf-8;base64,wrong","verification":{}},"mode":420}]},
+			"systemd":{}}`),
+			wantErrMsg: "illegal base64 data",
```

Comment:
at some point testing internal ignition error strings will get brittle and changes in the library will break the tests - the 3 error strings you have a probably fine though